catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java

```diff
@@ -91,44 +94,72 @@ public UpdateStorageRequest process(UpdateStorageRequest updateStorageRequest)
         return updateStorageRequest;
     }
 
+    private boolean isNitfMimeType(String rawMimeType) {
+        try {
+            return NitfInputTransformer.MIME_TYPE.match(rawMimeType);
+        } catch (MimeTypeParseException e) {
+            LOGGER.warn("unable to compare mime types: {} vs {}",
+                    NitfInputTransformer.MIME_TYPE,
+                    rawMimeType);
+        }
+        return false;
+    }
+
     private void process(List<ContentItem> contentItems) {
-        ContentItem contentItem = contentItems.get(0);
+        List<ContentItem> newContentItems = new LinkedList<>();
+        contentItems.forEach(contentItem -> process(contentItem).ifPresent(newContentItems::add));
+        contentItems.addAll(newContentItems);
+    }
+
+    private Optional<ContentItem> process(ContentItem contentItem) {
         Metacard metacard = contentItem.getMetacard();
 
+        if (!isNitfMimeType(contentItem.getMimeTypeRawData())) {
+            LOGGER.debug("skipping content item: filename={} mimeType={}",
+                    contentItem.getFilename(),
+                    contentItem.getMimeTypeRawData());
+            return Optional.empty();
+        }
+
         try {
             BufferedImage renderedImage = renderImage(contentItem);
 
             if (renderedImage != null) {
                 addThumbnailToMetacard(metacard, renderedImage);
-                ContentItem overviewContentItem = createOverview(contentItem.getId(), renderedImage,
+                ContentItem overviewContentItem = createOverview(contentItem.getId(),
+                        renderedImage,
                         metacard);
 
                 if (overviewContentItem != null) {
-                    contentItems.add(overviewContentItem);
+                    return Optional.of(overviewContentItem);
```

Comment:
Could you remove the conditional and just return 'Optional.ofNullable(overviewContentItem)'?